package/gluon-mesh-vpn-wireguard/src/key-translate.c

```diff
@@ -0,0 +1,69 @@
+#include <libubox/utils.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define FASTD_KEY_LEN_HEX 64
+
+static inline void invalid_key(void) {
+	fprintf(stderr, "Error: Invalid fastd key provided.\n");
+	exit(1);
+}
+
+static inline void usage(void) {
+	fprintf(stderr, "Usage: key-translate [-n] <fastd secret>\n");
+	fprintf(stderr, "-n     do not output the trailing newline\n");
+	exit(1);
+}
+
+static inline bool read_key(uint8_t key[32], const char *hexkey) {
+        if ((strlen(hexkey) != 64) || (strspn(hexkey, "0123456789abcdefABCDEF") != 64))
+                return false;
+
+        size_t i;
+        for (i = 0; i < 32; i++)
+                sscanf(&hexkey[2 * i], "%02hhx", &key[i]);
+
+        return true;
+}
+
+int main(int argc, char *argv[]) {
+	bool newline = true;
+	if (argc < 2 || argc > 3) {
+		usage();
+	} else if (argc==3) {
```

Comment:
`argc == 3`

Reply:
resolved in b626fcd73842859a8baa3c02bc1ae469a3250e6f